aa-sdk/ethers/src/__tests__/provider-adapter.test.ts

```diff
@@ -19,13 +26,26 @@ describe("Simple Account Tests", async () => {
 
   it("should correctly sign the message", async () => {
     const provider = await givenConnectedProvider({ alchemyProvider, signer });
+    const message = "hello world";
+    // todo: why does this provider's version of signMessage return a string, instead of Hex?
+    const signature = (await provider.signMessage(message)) as Hex;
+
+    // We must use a public client, rather than an account client, to verify the message, because AA-SDK incorrectly attaches the account address as a "from" field to all actions taken by that client, including the `eth_call` used internally by viem's signature verifier logic. Per EIP-684, contract creation reverts on non-zero nonce, and the `eth_call`'s from field implicitly increases the nonce of the account contract, causing the contract creation to revert.
+    const publicClient = createPublicClient({
+      chain: sepolia,
+      transport: http(`${endpoint}`),
+    });
+
     expect(
-      await provider.signMessage(
-        "0xa70d0af2ebb03a44dcd0714a8724f622e3ab876d0aa312f0ee04823285d6fb1b"
-      )
-    ).toBe(
-      "0x007ecc361d63ab82d89faeecfb79d40127f376c1ef3d545aeec3578eadce9d0c405a4d1ae6177bdebdc8413065014f735ee98da9643cc0e25c07a7423b694f8ae71b"
-    );
+      await publicClient.verifyMessage({
```

Comment:
I wrote this comment before realizing that the codebase is already doing things the way you did everywhere, so no need to change. But since I already wrote the comment, you get it anyways as a free FYI.

Nit: it's more idiomatic to write this assertion as
```ts
await expect(publicClient.verifyMessage({ ... })).resolves.toBe(true)
```
That way gives a clearer error message in the case where the assertion fails.

Likewise throughout.